bodhi/server/consumers/masher.py

```diff
@@ -256,14 +256,16 @@ def work(self, msg):
         last_key = None
         threads = []
         for batch in batches:
-            if last_key is not None and request_order_key(batch) != last_key:
+            if ((last_key is not None and request_order_key(batch) != last_key) or
+                    (len(threads) >= int(config.get('max_concurrent_mashes', 2)))):
                 # This means that after we submit all Stable+Security updates, we wait with kicking
                 # off the next series of mashes until that finishes.
                 self.log.info('All mashes for priority %s running, waiting', last_key)
```

Comment:
This log statement might not be entirely accurate now. Maybe we can set it to say something like this instead?

```
self.log.info('Waiting on %d mashes of priority %s.', len(threads), last_key)
```